QMLComponents/models/terms.cpp

```diff
@@ -223,67 +223,57 @@ Term &Terms::at(size_t index)
 	return _terms.at(index);
 }
 
-bool Terms::contains(const Term &term) const
+bool Terms::containsValue(const Term &term) const
 {
-	return std::find(_terms.begin(), _terms.end(), term) != _terms.end();
+	return containsValue(term.value());
 }
 
-bool Terms::contains(const std::string & component)
+bool Terms::containsValue(const QString &value) const
 {
-	return contains(tq(component));
+	for (const Term& term : _terms)
+		if (term.value() == value)
+			return true;
+
+	return false;
+}
+
+int Terms::indexOfValue(const Term &term) const
+{
+	return indexOfValue(term.value());
 }
 
-int Terms::indexOf(const QString &component) const
+int Terms::indexOfValue(const QString &value) const
 {
 	int i = 0;
-	for(const Term &term : _terms)
+	for (const Term& term : _terms)
 	{
-		if (term.contains(component))
+		if (term.value() == value)
```

Comment:
values are always unique because of the _makeUnique() I guess?

And isnt this a bit slow if there are a lot of terms?
Like, maybe its worth keeping some maps in Terms?

Reply:
They are unique, only for Paired VariablesList (using the ListModelMultiTermsAssigned model) a term can be used multiple times in their model. But this model does not use the Terms object, it uses its own 'Tuples' object instead.
It's a pity, because I could reuse the 'components' in the Term object to store these tuples, but this PR is already too big for the small feature I wanted to add....

Reply:
fair enough!

And it would have to be quite a lot of terms to get slow anyway